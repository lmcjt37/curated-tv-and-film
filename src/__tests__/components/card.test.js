import React from 'react';
import { render, screen } from '@testing-library/react';

import Card from '../../components/card';

const itemMock = {
  type: 'tv_show',
  title: 'Mock of Thrones',
  description: 'You know nothing',
  season: 1,
  episode: 1,
  episode_title: 'To kill a mockingbird',
  imdb: 'https://www.lmcjt.com',
  url: 'https://www.lmcjt.com',
  thumbnail: './assets/images/game-of-thrones.jpg',
  genre: ['Action', 'Adventure', 'Drama']
};

it('render all initial child components', () => {
  render(<Card {...itemMock} />);

  expect(screen.getByTestId('card')).toBeTruthy();
  expect(screen.getByTestId('card-media')).toBeTruthy();
  expect(screen.getByTestId('card-image')).toBeTruthy();
  expect(screen.getByTestId('card-content')).toBeTruthy();
  expect(screen.getByTestId('card-content-title')).toBeTruthy();
  expect(screen.getByTestId('card-content-subtitle-1')).toBeTruthy();
  expect(screen.getByTestId('card-content-subtitle-2')).toBeTruthy();
  expect(screen.getByTestId('card-content-chip-container')).toBeTruthy();
  expect(
    screen.getAllByTestId('card-content-chip-container-item').length
  ).toEqual(3);
  expect(screen.getByTestId('card-content-description')).toBeTruthy();
  expect(screen.getByTestId('card-content-actions')).toBeTruthy();
  expect(screen.getByTestId('card-action-imdb')).toBeTruthy();
  expect(screen.getByTestId('card-action-watch')).toBeTruthy();
});

it('snapshot of initial component', () => {
  const { asFragment } = render(<Card {...itemMock} />);
  expect(asFragment).toMatchSnapshot();
});
