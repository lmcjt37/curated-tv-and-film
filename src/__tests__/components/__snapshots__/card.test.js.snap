// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of initial component 1`] = `
<div
  class="MuiPaper-root MuiCard-root SingleCard-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    class="SingleCard-innerContainer-2"
  >
    <div
      class="MuiCardMedia-root SingleCard-media-3"
      style="background-image:url(\\"./assets/images/game-of-thrones.jpg\\")"
      title="Mock of Thrones"
    />
    <img
      alt="Mock of Thrones"
      class="SingleCard-thumbnail-4"
      src="./assets/images/game-of-thrones.jpg"
    />
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
      >
        Mock of Thrones 
      </h2>
      <div>
        <h3
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
        >
          To kill a mockingbird
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          Season: 1 | Episode: 1
        </p>
      </div>
      <div
        class="SingleCard-chips-5"
      >
        <div
          class="MuiChip-root SingleCard-chip-6"
        >
          <span
            class="MuiChip-label"
          >
            Action
          </span>
        </div>
        <div
          class="MuiChip-root SingleCard-chip-6"
        >
          <span
            class="MuiChip-label"
          >
            Adventure
          </span>
        </div>
        <div
          class="MuiChip-root SingleCard-chip-6"
        >
          <span
            class="MuiChip-label"
          >
            Drama
          </span>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        You know nothing
      </p>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <a
          aria-disabled="false"
          aria-label="IMDB"
          class="MuiButtonBase-root MuiFab-root SingleCard-fab-7 MuiFab-extended MuiFab-primary"
          href="https://www.lmcjt.com"
          rel="noopener noreferrer"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root SingleCard-localPlayIcon-9"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12c0-1.1.9-2 2-2V6c0-1.1-.9-2-2-2H4c-1.1 0-1.99.9-1.99 2v4c1.1 0 1.99.9 1.99 2s-.89 2-2 2v4c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2v-4c-1.1 0-2-.9-2-2zm-4.42 4.8L12 14.5l-3.58 2.3 1.08-4.12-3.29-2.69 4.24-.25L12 5.8l1.54 3.95 4.24.25-3.29 2.69 1.09 4.11z"
              />
            </svg>
            IMDB
          </span>
        </a>
        <a
          aria-disabled="false"
          aria-label="Watch"
          class="MuiButtonBase-root MuiFab-root SingleCard-fab-7 MuiFab-extended MuiFab-primary"
          href="https://www.lmcjt.com"
          rel="noopener noreferrer"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root SingleCard-playCircleOutlineIcon-8"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 16.5l6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
            Watch
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
